src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IIsPatternExpression.cs

```diff
@@ -797,7 +856,42 @@ void M(int? x1, int x2, bool b)
             VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedFlowGraph, expectedDiagnostics);
         }
 
-        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow, CompilerFeature.Patterns)]
+        [Fact]
+        public void IsPattern_RecursivePattern()
+        {
+            string source = @"
+class C
+{
+    void M((int X, int Y) tuple, bool b)
+    {
+        if (/*<bind>*/tuple is (1, 2) { Item1: int x } y/*</bind>*/) { }
+    }
+}
+";
+            string expectedOperationTree = @"
+IIsPatternOperation (OperationKind.IsPattern, Type: System.Boolean) (Syntax: 'tuple is (1 ... : int x } y')
+  Expression: 
+    IParameterReferenceOperation: tuple (OperationKind.ParameterReference, Type: (System.Int32 X, System.Int32 Y)) (Syntax: 'tuple')
+  Pattern: 
+    IRecursivePatternOperation (Declared Symbol: (System.Int32 X, System.Int32 Y) y, MatchedType: (System.Int32 X, System.Int32 Y), Deconstruct Symbol: null)
```

Comment:
>(System.Int32 X, System.Int32 Y) [](start = 49, length = 32)

The declared symbol having tuple names feels unexpected. Does that reflect the language in some way?
I would expect that `y` is an `(int, int)`.
Did we discuss allowing names in positional patterns? I'd expect the names in output to come from there `tuple is (a: 1, b: 2) y`, rather than from input.

Reply:
If the type is not given in the pattern, the type of the variable is its input type.  Using `Deconstruct` would not change that (it does not produce a tuple type).

---
In reply to: [244035791](https://github.com/dotnet/roslyn/pull/31967#discussion_r244035791) [](ancestors = 244035791)